devito/operations/interpolators.py

```diff
@@ -108,15 +107,12 @@ def interpolate(self, *args, **kwargs):
         pass
 
 
-class LinearInterpolator(GenericInterpolator):
+class WeightedInterpolation(GenericInterpolator):
```

Comment:
For homogeneity with the other classes I would rename it as WeightedInterpolator.
Unless you want to rename the other classes to xxxInterpolation..?